pkg/kubeapiserver/default_storage_factory_builder.go

```diff
@@ -89,6 +89,7 @@ func mergeAPIResourceConfigs(defaultAPIResourceConfig *serverstorage.ResourceCon
 		} else if allAPIFlagValue == "true" {
 			resourceConfig.EnableVersions(legacyscheme.Registry.RegisteredGroupVersions()...)
 		}
+		delete(overrides, "api/all")
```

Comment:
Why removing it from `overrides`? Same as below.
No need to do that IMO.

`overrides` will be used in below `group/version/resource` as well.

Also it's quite dangerous to remove a element from the list, especially during iteration.

Reply:
>Why removing it from overrides? Same as below.

Only pass `runtime-config` not used by previous apiservers.

>overrides will be used in below group/version/resource as well.
only delete that used.

I search https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-golang-map-within-a-range-loop
and find it is safe, in java it is unsafe .

Reply:
>overrides will be used in below group/version/resource as well.

does not matter, only delete which used already.

Reply:
if you delete it here, only the ones registered into the legacy scheme will be enabled... don't we need to leave this in place for delegates to see that all are desired as well?

Reply:
@liggitt You mean this `api/all` also allow all other apiserver's registry? If so, this should be reserved.

Reply:
I found, we can not delete this here.  It is called twice in kube-apiserver.